android-offline-maps/app/src/main/java/com/example/mapbox_offline/TileRegionDownloadView.kt

```diff
@@ -0,0 +1,151 @@
+
+import android.util.Log
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.CheckCircle
+import androidx.compose.material3.Button
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Divider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.mapbox.common.MapboxOptions
+import com.mapbox.common.TileStore
+import com.mapbox.common.TileRegion
+import com.mapbox.maps.mapsOptions
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TileRegionDownloadScreen(
+    viewModel: TileRegionDownloadViewModel = TileRegionDownloadViewModel(),
+    onDone: () -> Unit = {}
+) {
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                navigationIcon = {
+                    IconButton(onClick = onDone) {
+                        Icon(Icons.Filled.ArrowBack, contentDescription = "Back")
+                    }
+                },
+                title = { Text("Offline Regions") },
+                actions = {
+                    TextButton(onClick = { viewModel.clearAllRegions() }) {
+                        Text("Clear All", color = MaterialTheme.colorScheme.error)
+                    }
+                }
+            )
+        }
+    ) { padding ->
+        LazyColumn(modifier = Modifier
+            .fillMaxSize()
+            .padding(padding)
+            .padding(8.dp)
+        ) {
+            items(viewModel.regions) { region ->
+                RegionRow(
+                    region = region,
+                    isDownloading = viewModel.downloadingRegions.contains(region.id),
+                    progress = viewModel.downloadProgress[region.id] ?: 0f,
+                    refreshTrigger = viewModel.refreshTrigger,
+                    onDownload = { viewModel.downloadRegion(region) }
+                )
+                Divider()
+            }
+        }
+    }
+}
+
+@Composable
+fun RegionRow(
+    region: OfflineRegion,
+    isDownloading: Boolean,
+    progress: Float,
+    refreshTrigger: Boolean,
+    onDownload: () -> Unit
+) {
+    var isDownloaded by remember { mutableStateOf(false) }
+    var sizeInMB by remember { mutableStateOf(0.0) }
+    val TAG = "RegionRow-${region.id}"
+    val tileStore: TileStore? = MapboxOptions.mapsOptions.tileStore!!
```

Comment:
Same as above, not null assertion is not needed I think
```suggestion
    val tileStore: TileStore? = MapboxOptions.mapsOptions.tileStore
```